packages/metadata/src/types.ts

```diff
@@ -0,0 +1,90 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: @loopback/metadata
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT'
+
+/**
+ * Decorator function types
+ */
+export type DecoratorType =
+  | ClassDecorator
+  | PropertyDecorator
+  | MethodDecorator
+  | ParameterDecorator;
+
+/**
+ * A strongly-typed metadata accessor via reflection
+ * @typeparam T Type of the metadata value
+ * @typeparam D Type of the decorator
+ */
+export class MetadataAccessor<T, D extends DecoratorType = DecoratorType> {
```

Comment:
The class shouldn't be in a `types.ts` file. Should have it's own file.

Reply:
I know a class serves as a type but just thought `types.ts` for was types / interfaces not classes.

Didn't see the circular dependency. Good enough reason to keep it here.